src/Microsoft.Health.Fhir.Core/Features/Conformance/Serialization/ProfileReferenceConverter.cs

```diff
@@ -0,0 +1,62 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using EnsureThat;
+using Microsoft.Health.Fhir.Core.Models;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+
+namespace Microsoft.Health.Fhir.Core.Features.Conformance.Serialization
+{
+    internal class ProfileReferenceConverter : JsonConverter
+    {
+        private IModelInfoProvider _modelInfoProvider;
+
+        public ProfileReferenceConverter(IModelInfoProvider modelInfoProvider)
+        {
+            _modelInfoProvider = modelInfoProvider;
+        }
+
+        public override bool CanRead => false;
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            EnsureArg.IsNotNull(writer, nameof(writer));
+            EnsureArg.IsNotNull(serializer, nameof(serializer));
+
+            if (value is ReferenceComponent obj)
+            {
+                if (_modelInfoProvider.Version.Equals(FhirSpecification.Stu3))
+                {
+                    serializer = new JsonSerializer()
+                    {
+                        ContractResolver = new CamelCasePropertyNamesContractResolver(),
+                    };
```

Comment:
Define this serializer in the constructor, shouldn't need to override the parameter